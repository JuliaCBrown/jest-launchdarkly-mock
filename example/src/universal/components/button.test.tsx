import React from 'react'
import { render } from '@testing-library/react'
// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import { mockFlags } from '../../../ld-mock.js'
import Button from './button'

describe('button', () => {
  it('flag on', () => {
    mockFlags({ 'dev-test-flag': true })
    const { asFragment } = render(<Button />)
    expect(asFragment()).toMatchSnapshot()
  })

  // it('flag off', () => {
  //   mockFlags({ 'dev-test-flag': false })
  //   const { asFragment } = render(<Button />)
  //   expect(asFragment()).toMatchSnapshot()
  // })
})
