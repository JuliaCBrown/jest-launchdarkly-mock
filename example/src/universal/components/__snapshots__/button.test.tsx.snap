// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button flag off 1`] = `
<DocumentFragment>
  Button is disabled
</DocumentFragment>
`;

exports[`button flag on 1`] = `
<DocumentFragment>
  <button />
</DocumentFragment>
`;
